1-Draft/RFC0000-Chain-Operators.md

```diff
@@ -0,0 +1,611 @@
+---
+RFC:
+Author: Robert Holt
+Status: Draft
+Area: Pipeline Chain Operators
+Comments Due: 2019-07-12 0000
+Plan to implement: Yes
+---
+
+# Pipeline Chain Operators
+
+POSIX shells have what may be referred to as *AND-OR lists*,
+what the [Open Group's Shell Command Language specification](https://pubs.opengroup.org/onlinepubs/007904875/utilities/xcu_chap02.html#tag_02_09_03)
+describes as:
+
+> a sequence of one or more pipelines separated by the operators `&&` and `||`
+
+Semantically, these operators look at the exit code of the left-hand side pipeline
+and conditionally execute the right-hand side pipeline.
+`&&` executes the right-hand side if the left-hand side "succeeded",
+and `||` executes the right-hand side if the left-hand side did not "succeed".
+
+Some examples in bash:
+
+```bash
+echo 'Success' && echo 'Second success' # Executes both left-hand and right-hand command
+echo 'Success' || echo 'Second success' # Executes only the left-hand command
+false && echo 'Second success' # Executes only the left-hand command (false always "fails")
+false || echo 'Second success' # Executes both left-hand and right-hand commands
+false && echo foo || echo bar # Writes only "bar" to stdout (operators are left-associative)
+true || echo foo && echo bar # Writes only "bar" to stdout
+```
+
+Also from the Shell Command Language specification:
+
+> A ';' or &lt;newline&gt; terminator shall cause the preceding AND-OR list to be executed sequentially;
+> an '&' shall cause asynchronous execution of the preceding AND-OR list.
+
+Meaning that the entire AND-OR list is backgrounded:
+
+```bash
+true && echo 'Success' & # The list `true && echo 'Success'` is executed in the background
+```
+
+This RFC proposes adding AND-OR lists to PowerShell, using the same `&&` and `||`.
+Because "AND-OR list" is neither intuitive nor common as a terminology,
+instead the term **pipeline chains** is proposed.

```

Comment:
Basically because the proposed implementation is for the operators to be interspersed between pipelines. They are syntactically related to the PipelineAst:

```powershell
1,2,3 | % { Write-Output $_ } && Write-Host "Succeeded"
```

Command chain operator doesn't illustrate the fact that you can slide a whole pipeline between them.